src/test/ui/impl-trait/impl-fn-hrtb-bounds.stderr

```diff
@@ -22,30 +22,57 @@ note: lifetime declared here
 LL | fn a() -> impl Fn(&u8) -> (impl Debug + '_) {
    |                   ^
 
+error: lifetime may not live long enough
+  --> $DIR/impl-fn-hrtb-bounds.rs:6:9
+   |
+LL |     |x| x
+   |      -- ^ returning this value requires that `'1` must outlive `'2`
+   |      ||
+   |      |return type of closure is impl Debug + '2
+   |      has type `&'1 u8`
```

Comment:
Could we somehow silence this? (Can be follow up work, but do we have a clear way to do it?)

Reply:
I don't really know why the compiler does not abort any more. I speculate that the `higher kinded lifetime bounds on nested opaque types are not supported yet` error is not emitted from the same code path any more, so the `track_errors` calls in `rustc_hir_analysis::check_crate` do not trigger any more.

Reply:
I added a commit, which should remove some spurious errors.